netbox/extras/api/serializers.py

```diff
@@ -454,7 +454,7 @@ class ConfigTemplateSerializer(TaggableModelSerializer, ValidatedModelSerializer
         required=False
     )
     data_file = NestedDataFileSerializer(
-        read_only=True
+        read_only=False
```

Comment:
It usually defaults to false is not specified, so this can be removed entirely

Reply:
I think I should put `required=False`, because `data_file` parameter is not mandatory, as for `data_source` just above ? I suppose it has been set because default value for `required` field is True ?

Reply:
That's confirmed by the CI:
```
AssertionError: 400 != 201 : Expected HTTP status 201; received 400: {'data_file': [ErrorDetail(string='This field is required.', code='required')]}
```
I'll send an update soon.